src/commands/createNewProject/ProjectCreateStep/PowerShellProjectCreateStep.ts

```diff
@@ -97,43 +97,24 @@ export class PowerShellProjectCreateStep extends ScriptProjectCreateStep {
         });
 
         try {
-            const xmlResult: string = await this.getPSGalleryAzModuleInfo();
-            const versionResult: string = await this.parseLatestAzModuleVersion(xmlResult);
+            const response: HttpOperationResponse = await requestUtils.sendRequestWithTimeout({ method: 'GET', url: this.azModuleGalleryUrl });
+            const versionResult: string = await this.parseLatestAzModuleVersion(response);
             const [major]: string[] = versionResult.split('.');
             return parseInt(major);
         } catch {
             return undefined;
         }
     }
 
-    private async getPSGalleryAzModuleInfo(): Promise<string> {
-        const request: requestUtils.Request = await requestUtils.getDefaultRequestWithTimeout(this.azModuleGalleryUrl, undefined, 'GET');
-        return await requestUtils.sendRequest(request);
-    }
-
-    private async parseLatestAzModuleVersion(azModuleInfo: string): Promise<string> {
-        const moduleInfo: string = await new Promise((
-            resolve: (ret: string) => void,
-            // tslint:disable-next-line:no-any
-            rejects: (reason: any) => void): void => {
-            // tslint:disable-next-line:no-any
-            xml2js.parseString(azModuleInfo, { explicitArray: false }, (err: any, result: string): void => {
-                if (err) {
-                    rejects(err);
-                } else {
-                    resolve(result);
-                }
-            });
-        });
-
-        // tslint:disable-next-line:no-string-literal no-unsafe-any
-        if (moduleInfo['feed'] && moduleInfo['feed']['entry'] && Array.isArray(moduleInfo['feed']['entry'])) {
```

Comment:
How come you were able to get rid of this logic completely?

Reply:
The new sdk will automatically parse xml and return it in `response.parsedBody`, so I changed to use that instead of manually parsing with "xml2js" ourselves. When I went to test it, the parsed object was slightly different, not entirely sure why. The new sdk still uses "xml2js" under the covers, but maybe they pass in different options?